File: gm-utils/src/main/java/src/commom/utils/string/StringBeforeLast.java
package src.commom.utils.string;

import src.commom.utils.integer.IntegerCompare;

public class StringBeforeLast {

	private StringBeforeLast() {}

	public static String get(String s, String substring) {
		int i = s.lastIndexOf(substring);
		if (IntegerCompare.eq(i, -1)) {
			return null;
		}
		return s.substring(0, i);
	}

	public static String obrig(String s, String substring) {
		return StringObrig.get(get(s, substring));
	}

}

File: gm-utils/js/src/commom/utils/string/StringBeforeLast.js
import IntegerCompare from '../integer/IntegerCompare';
import StringObrig from './StringObrig';

export default class StringBeforeLast {

	static get(s, substring) {
		let i = s.lastIndexOf(substring);
		if (IntegerCompare.eq(i, -1)) {
			return null;
		}
		return s.substring(0, i);
	}

	static obrig(s, substring) {
		return StringObrig.get(StringBeforeLast.get(s, substring));
	}

}
